library/core/src/ptr/mod.rs

```diff
@@ -714,7 +715,8 @@ pub const fn without_provenance_mut<T>(addr: usize) -> *mut T {
 #[rustc_const_stable(feature = "stable_things_using_strict_provenance", since = "1.61.0")]
 #[unstable(feature = "strict_provenance", issue = "95228")]
 pub const fn dangling_mut<T>() -> *mut T {
-    without_provenance_mut(mem::align_of::<T>())
+    // 0x8000[...], aligned for any pointer.
+    without_provenance_mut(isize::MIN as usize)
```

Comment:
makes sense.

another thing that probably should be done is using a 32 bit value instead of a 64 bit one if possible (the asm for that is shorter, as-is the perf regressions are *probably*(?) due to using larger instructions). don't think we even support larger than 32 bit alignments, do we?

will poke at that when i'm more free this week

Reply:
alignments can be at most `2^29` currently

Reply:
as far as I can see (looking at godbolt), it's treated as just a literal? though ofc inlining might affect the code from here

```rust
#[no_mangle]
pub fn return_16() -> *const u8 {
    (1_usize<<16) as *const u8
}

#[no_mangle]
pub fn return_48() -> *const u8 {
    (1_usize<<48) as *const u8
}
```

```
return_16:
 mov    eax,0x10000
 ret

return_48:
 movabs rax,0x1000000000000
 ret
```

https://godbolt.org/z/b4Tb5Pzdo

(with `movabs` being a larger instruction)